doc/source/nibabel_images.rst

```diff
@@ -220,21 +220,38 @@ False
 Getting the image data the easy way
 ===================================
 
-For either type of image (array or proxy) you can always get the data with
-the :meth:`get_data() <nibabel.spatialimages.SpatialImage.get_data>` method.
+For either type of image (array or proxy) you can always get the data with the
+:meth:`get_fdata() <nibabel.spatialimages.SpatialImage.get_fdata>` method.
 
-For the array image, ``get_data()`` just returns the data array:
+For the array image, ``get_fdata()`` just returns the data array, if it's already the required floating point type (default 64-bit float).  If it isn't that type, ``get_fdata()`` casts it to one:
 
->>> image_data = array_img.get_data()
+>>> image_data = array_img.get_fdata()
 >>> image_data.shape
 (2, 3, 4)
->>> image_data is array_data
+>>> image_data.dtype == np.dtype(np.float64)
 True
 
-For the proxy image, the ``get_data()`` method fetches the array data from
+The cast to floating point means the array is not the one attached to the image:
+
+>>> image_data is array_img.dataobj
+False
+
+Here's an image backed by a floating point array:
+
+>>> farray_img = nib.Nifti1Image(image_data.astype(np.float64), affine)
+>>> farray_data = farray_img.get_fdata()
+>>> farray_data.dtype == np.dtype(np.float64)
+True
+
+There was no cast, so the array returned is exactly the array attached to the image:
+
+>>> image_data is array_img.dataobj
+False
```

Comment:
I might be misreading, but from the above sentence, I'd think this ought to be:

```Python
>>> farray_data is farray_img.dataobj
True
```

Reply:
Good spot - you're right, but then this test fails, revealing the bug I think I have fixed in https://github.com/nipy/nibabel/pull/638